sunbird-rc-esignet-integration-impl/src/main/java/io/mosip/esignet/sunbirdrc/integration/service/SunbirdRCVCIssuancePlugin.java

```diff
@@ -171,6 +187,40 @@ private Map<String,Object> fetchRegistryObject(String entityUrl) throws VCIExcha
         }
     }
 
+    private Map<String, Object> fetchRegistryObjectByPSUT(String registrySearchUrl, String psut) throws VCIExchangeException {
+
+        RegistrySearchRequestDto registrySearchRequestDto=new RegistrySearchRequestDto();
+        registrySearchRequestDto.setOffset(0);
+        registrySearchRequestDto.setLimit(2);
+        Map<String,Map<String,String>> filter=new HashMap<>();
+        Map<String,String> fullName=new HashMap<>();
+        fullName.put(FILTER_EQUALS_OPERATOR,psut);
+        filter.put(PSUT_TOKEN,fullName);
+        registrySearchRequestDto.setFilters(filter);
+
+        RequestEntity requestEntity =RequestEntity.post(UriComponentsBuilder.fromUriString(registrySearchUrl).build().toUri())
+                .contentType(MediaType.APPLICATION_JSON_UTF8)
+                .body(registrySearchRequestDto);
+
+        ResponseEntity<List<Map<String,Object>>> responseEntity = restTemplate.exchange(requestEntity,
+                new ParameterizedTypeReference<List<Map<String,Object>>>() {});
+        if (responseEntity.getStatusCode().is2xxSuccessful() && responseEntity.getBody() != null) {
+            List<Map<String,Object>> responseList = responseEntity.getBody();
+            if(responseList.size()==1){
+                //TODO  This need to be removed since it can contain PII
+                log.debug("getting response {}", responseEntity);
+                return responseEntity.getBody().get(0);
+            }else{
+                log.error("Registry search returns more than one match, so authentication is considered as failed. Result size: " + responseList.size());
+                throw new VCIExchangeException(ErrorConstants.VCI_EXCHANGE_FAILED );
```

Comment:
In this case if there is more than 1, possible to add a debug log that is has more then then take the first element ? Also add TODO on this method that we should find a generic way to sort desc based on created date. The idea is we give out the most recent record as VC. But for later.

Reply:
added the required infomation